### DISCUSSION TITLE
generate migration on existing database

### DISCUSSION THREAD

**Discussion by mradulsethi05** [prisma/prisma#11000] [Q&A] [answered]
I have a existing database for application, that was an on premise database. I am not using all tables in my current application. when i add new models in my application and try to migrate that model. It will give me an warning to lost all the existing table from database, but i can not remove that table from a database (It's an production database). 

I have tried possible solution -

1. I will took a pull from an existing database, It will clone a tables in a prisma.schema file when i run a migration command it will give me a warning of data lost.
2. I used a local database and create a migration files and deploy it on production database it gives me same warning .
3. I run a command **prisma migrate dev --create-only**  and push it on production same issue.

I am also OK with wrote a sql command and it will make changes on the database.

**Comment by nurul3101** [comment, marked as answer]
Hey @mradul070 👋 

You would need to [Baseline](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining) your database.
Baselining tells Prisma Migrate to assume that one or more migrations have already been applied. It's useful for databases that contain data(like production), which means that the database cannot be reset.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
